diff --git a/cmake/QtBuildPathsHelpers.cmake b/cmake/QtBuildPathsHelpers.cmake
index bf28d1e9..0d7dd036 100644
--- a/cmake/QtBuildPathsHelpers.cmake
+++ b/cmake/QtBuildPathsHelpers.cmake
@@ -141,8 +141,9 @@ function(qt_configure_process_path name default docstring)
     elseif(rel_path MATCHES "^\.\./")
         # INSTALL_SYSCONFDIR is allowed to be outside the prefix.
         if(NOT name STREQUAL "INSTALL_SYSCONFDIR")
-            message(FATAL_ERROR
-                "Path component '${name}' is outside computed install prefix: ${rel_path} ")
+            set(new_value "${rel_path}")
+        else()
+            set(new_value "${${name}}")
         endif()
         # Keep the absolute path.
         set(new_value "${${name}}")
